backend/infrahub/git/base.py

```diff
@@ -500,8 +503,20 @@ def get_branches_from_remote(self) -> dict[str, BranchInRemote]:
             if short_name == "HEAD":
                 continue
 
+            if config.SETTINGS.git.import_sync_branch_names and not branch_name_in_sync_branches(
+                branch_short_name=short_name
+            ):
+                skipped_branch_names.append(short_name)
+                continue
```

Comment:
Here we must also account for the default branch, any branch in Infrahub marked as "sync with git", and potentially any "default branch" configured on the repo itself (this is typically always "main" but it can differ).